src/misc/ethernet_pcap.h

```diff
@@ -0,0 +1,47 @@
+/* *  Copyright (C) 2020  The DOSBox Team
+ *
+ *  This program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License along
+ *  with this program; if not, write to the Free Software Foundation, Inc.,
+ *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+
+#ifndef DOSBOX_ETHERNET_PCAP_H
+#define DOSBOX_ETHERNET_PCAP_H
+
+#include "config.h"
+
+#if C_PCAP
+
+#include "ethernet.h"
+
+#ifdef WIN32
+#define HAVE_REMOTE
+#endif
+
+#include "support.h" /* Prevents snprintf conflict with pcap on Windows */
+#include "pcap.h"
+
+class PcapEthernetConnection : public EthernetConnection {
+	public:
+		PcapEthernetConnection(void);
+		~PcapEthernetConnection(void);
+		bool Initialize(void);
+		void SendPacket(uint8_t* packet, int len);
+		void GetPackets(std::function<void(uint8_t*, int)> callback);
+	private:
+		pcap_t *adhandle = 0;
```

Comment:
`0` -> `nullptr`

Reply:
> I think NULL and nullptr may be different in some cases, no?

It's a safer drop-in replacement for `NULL` with all the same runtime behavior, but the safety come with compile-time checks.  Definitely recommended (and can be used when interfacing w/ old C libraries without issue), see: 
 
https://embeddedartistry.com/blog/2017/03/08/migrating-from-c-to-c-null-vs-nullptr/